pkg/cli/tail.go

```diff
@@ -176,21 +200,50 @@ func (c *TailCommand) Run(ctx context.Context, args []string) error {
 		return fmt.Errorf("-max-num must be greater than 0 and less than 1000")
 	}
 
-	// Tail logs.
-	ls, err := c.tail(ctx)
+	var extra []validation.Validator
+	if c.flagAdditionalCheck {
+		extra = append(extra, validation.ValidateLabels)
+	}
+
+	var puller logPuller
+	if c.testPuller != nil {
+		puller = c.testPuller
+	} else {
+		logClient, err := logging.NewClient(ctx)
+		defer func() {
+			if err := logClient.Close(); err != nil {
+				rErr = errors.Join(rErr, fmt.Errorf("failed to close log client: %w", err))
+			}
+		}()
+		if err != nil {
+			return errors.Join(rErr, fmt.Errorf("failed to create logging client: %w", err))
+		}
+		puller = cloudlogging.NewPuller(ctx, logClient, c.flagScope, cloudlogging.WithRetry(retry.WithMaxRetries(uint64(c.flagFollowDuration), retry.NewConstant(1000*time.Millisecond))))
```

Comment:
Any issue with the default retry? is this for using flagFollowDuration?

Reply:
The default retry ends ofter 5 retries, and stream will do a retry if client failed to send request, base on experiments, it's very easy to reach the limit.
I have updated the code so `list` still uses the default retry, but `stream` uses another retry.